jerry-core/ecma/operations/ecma-proxy-object.c

```diff
@@ -897,9 +897,236 @@ ecma_collection_t *
+
+  /* 12. */
+  if (ECMA_IS_VALUE_ERROR (extensible_target))
+  {
+    ecma_collection_free (trap_result);
+    return NULL;
+  }
+
+  /* 13. */
+  ecma_collection_t *target_keys = ecma_op_object_get_property_names (target_obj_p, ECMA_LIST_SYMBOLS);
+
+  /* 14. */
+  if (target_keys == NULL)
+  {
+    ecma_collection_free (trap_result);
+    return target_keys;
+  }
+
+  /* 16. */
+  ecma_collection_t *target_configurable_keys = ecma_new_collection ();
+
+  /* 17. */
+  ecma_collection_t *target_non_configurable_keys = ecma_new_collection ();
+
+  ecma_collection_t *ret_value = NULL;
+
+  /* 18. */
+  for (uint32_t i = 0; i < target_keys->item_count; i++)
+  {
+    ecma_property_descriptor_t target_desc;
+
+    ecma_string_t *prop_name_p = ecma_get_prop_name_from_value (target_keys->buffer_p[i]);
+
+    ecma_value_t status = ecma_op_object_get_own_property_descriptor (target_obj_p,
+                                                                      prop_name_p,
+                                                                      &target_desc);
+
+    if (ECMA_IS_VALUE_ERROR (status))
+    {
+      ecma_collection_free (trap_result);
+      goto free_collections;
+    }
+
+    ecma_value_t prop_value = ecma_make_prop_name_value (prop_name_p);
+
+    if (ecma_is_value_true (status)
+        && !(target_desc.flags & ECMA_PROP_IS_CONFIGURABLE))
+    {
+      ecma_collection_push_back (target_non_configurable_keys, prop_value);
+    }
+    else
+    {
+      ecma_collection_push_back (target_configurable_keys, prop_value);
+    }
+
+    if (ecma_is_value_true (status))
+    {
+      ecma_free_property_descriptor (&target_desc);
+    }
+  }
+
+  /* 19. */
+  if (ecma_is_value_true (extensible_target) && target_non_configurable_keys->item_count == 0)
+  {
+    ret_value = trap_result;
+    goto free_collections;
+  }
+
+  /* 20. */
+  ecma_collection_t *unchecked_result_keys = ecma_new_collection ();
+
+  ecma_collection_append (unchecked_result_keys, trap_result->buffer_p, trap_result->item_count);
+
+  for (uint32_t i = 0; i < unchecked_result_keys->item_count; i++)
+  {
+    ecma_string_t *unchecked_prop_name = ecma_get_prop_name_from_value (unchecked_result_keys->buffer_p[i]);
+    ecma_ref_ecma_string (unchecked_prop_name);
+  }
+
+  /* 21. */
+  for (uint32_t i = 0; i < target_non_configurable_keys->item_count; i++)
+  {
+    ecma_string_t *current_prop_name = ecma_get_prop_name_from_value (target_non_configurable_keys->buffer_p[i]);
+
+    uint32_t index = UINT32_MAX;
+
+    for (uint32_t j = 0; j < unchecked_result_keys->item_count; j++)
+    {
+      ecma_string_t *unchecked_prop_name = ecma_get_prop_name_from_value (unchecked_result_keys->buffer_p[j]);
+
+      if (ecma_compare_ecma_strings (current_prop_name, unchecked_prop_name))
+      {
+        if (index == UINT32_MAX)
+        {
+          index = j;
+        }
+        else
+        {
+          ecma_collection_remove (unchecked_result_keys, j);
```

Comment:
I was wondering if it would be possible to not remove the keys in each step (if found), just make the given position empty (replace it with an empty value). At the end a simple check if there is any non-empty value would give us the result that there is an error somewhere. What do you think @rerobika ?

Reply:
maybe it could work, I support to try it.